test/components/componentTests.ts

```diff
@@ -78,6 +78,17 @@ describe("Component", () => {
       svg.remove();
     });
 
+    it("remove dom elements in previous svg element when switching element", () => {
+      c.anchor(svg);
+      let svg2 = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      c.anchor(svg2);
+      assert.isFalse((<Element>svg.node()).hasChildNodes(), "previous svg element should not have any childnodes");
+      assert.isTrue((<Element>svg2.node()).hasChildNodes(), "new svg element should have childnodes.");
```

Comment:
I think I have tried `hasChildNodes()` before with sad times from PhantomJS.  Alternatively, you can try `selection.empty()`